lib/common.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:flutter/material.dart';
+
+// Suppose this comes from a package and cannot be known by any
+// libraries we invent (and doesn't know about our libraries).
+mixin ShowAnimationsMixin<T extends StatefulWidget> on State<T> {
+  bool get showAnimations => _showAnimations;
+  bool _showAnimations = true;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
```

Comment:
Problem is that the mixin's definition of didChangeDependencies replaces the StatefulAnimationState's definition. So you can't separate the logic out like this using stateful widgets. This is one of the issues that is addressed by hooks.

Reply:
I'm not sure I follow what you mean by "replaces". It's just inheritance, they all call each other, no?

Reply:
Forgive me. In this case it is just inheritance. I was thinking of multiple mixins and the problem where one mixin's definition would shadow the other mixin's definition.